driver hwmon adt7475 clear pwm invert bit

The Accton 4654, the only system using this driver, has a problem
where sometimes the power on value of the PWM control registers
(registers 0x5C,0x5D,0x5E) differs from the default.  Sometimes the
"PWM invert" bit is set incorrectly, it should be clear for normal
operations.

With the improper setting the fans will not spin.

This patch clears the "PWM invert" bit every time the PWM value is
updated via sysfs.

Clear PWM invert bit, i.e. force normal sense of duty cycle.  See
ADT7473 data sheet for description of register 0x5C, bit 4:

  This bit inverts the PWM output. The default is 0, which corresponds
  to a logic high output for 100% duty cycle. Setting this bit to 1
  inverts the PWM output, so 100% duty cycle corresponds to a logic
  low output.

diff --git a/drivers/hwmon/adt7475.c b/drivers/hwmon/adt7475.c
index b5fcd87..d460f4e 100644
--- a/drivers/hwmon/adt7475.c
+++ b/drivers/hwmon/adt7475.c
@@ -671,6 +671,21 @@ static ssize_t set_pwm(struct device *dev, struct device_attribute *attr,
 			return count;
 		}
 
+		/* Clear PWM invert bit, i.e. force normal sense of
+		 * duty cycle.  See ADT7473 data sheet for description
+		 * of register 0x5C, bit 4:
+		 *
+		 *   This bit inverts the PWM output. The default is
+		 *   0, which corresponds to a logic high output for
+		 *   100% duty cycle. Setting this bit to 1 inverts
+		 *   the PWM output, so 100% duty cycle corresponds to
+		 *   a logic low output.
+		 *
+		 */
+		data->pwm[CONTROL][sattr->index] &= ~(1 << 4);
+		i2c_smbus_write_byte_data(client, PWM_CONFIG_REG(sattr->index),
+					  data->pwm[CONTROL][sattr->index]);
+
 		reg = PWM_REG(sattr->index);
 		break;
 
